picture.c

```diff
@@ -1116,7 +1116,6 @@ void put_picture(struct context *cnt, char *file, unsigned char *image, int ftyp
     put_picture_fd(cnt, picture, image, cnt->conf.quality, ftype);
 
     myfclose(picture);
-    event(cnt, EVENT_FILECREATE, NULL, file, (void *)(unsigned long)ftype, NULL);
```

Comment:
It would seem that we would need to replace this call in picture.c with other calls from the mlp routines.

Reply:
This event call was moved in the first patch from picture.c->put_picture() to the file event.c-> event_image_detect() to have the currenttime_tv available.
(perhaps i should have made it more clearly in the commit message or make it a separate patch)
It's to exactly clear to me what you mean with "replace this call in picture.c with other calls from the mlp routines" ?

Reply:
First, I fully support getting that call to `EVENT_FILECREATE` out of the picture.c module.  It isn't an appropriate place for making that call.  

The `mlp` routines I was referring to are all the functions within the `motion.c` module that are primarily responsible for calling the event routines as pictures are processed.  

I have not reviewed all the flows of the program with respect to this change but was thinking that somewhere in one of the mlp functions could be a "central" location in which a call to the `EVENT_FILECREATE` could be placed.  

The general problem is that it appears with this change, the `EVENT_FILECREATE` is only called on certain execution paths.  In the previous code, with the call being in picture.c, I believe the `EVENT_FILECREATE` would have been called indirectly in the [`event_imagem_detect`](https://github.com/sander1234567890/motion/blob/cc64586dd2b8d0b7e5a52099a22369254693d7e5/event.c#L447)  as well as from the `event_image_snapshot` and probably other functions since they are calling `put_picture`.  

If there isn't a central spot to put the call to `EVENT_FILECREATE` in a mlp_ function, then I think that calls would need to placed into the event.c next to all calls to `put_picture` so that we maintain the current events being triggered.

Reply:
As far as i can tell there isn't a call to put_picture() from motion.c.
I see there are some more calls site in event.c in functions like event_image_snapshot(), event_imagem_detect() etc.
I will put the calls to EVENT_FILECREATE there as well.